### ISSUE TITLE
Some word cannot be used as field of properties in object definition

### ISSUE THREAD

**Issue by lubronzhan** [go-swagger/go-swagger#704]
## Problem statement

Some alphabet could not be used as field in object definitions properties.
## Steps to reproduce

I tried to make a object in definition section in swagger.yml, like this

```
definitions: 
  key:
    description: key content
    properties:
      kty:
        description: type
        type: string
      e:
        description: key
        type: string
      use:
        description: sign
        type: string
      alg:
        description: algorithm
        type: string
      n:
        description: key
        type: string
```

But it reports error when validate the swagger.yml

```
swagger validate ./swagger/user_service.yml                                                                                                                                                        
2016/10/20 11:54:54 types don't match: expect map key string or int get: bool                                                                                                                      
make: *** [swagger-validate] Error 1    
```

After I remove the 

```
      n: 
        description: key
        type: string
```

Then there is no error.

Some alphabets works, some not.

**Comment by casualjim** [comment]
This is because a property name of n in yaml turns into a boolean. 

When I use the yaml from the issue I get:

```go
map[interface {}]interface {}{
    "definitions": map[interface {}]interface {}{
        "key": map[interface {}]interface {}{
            "description": "key content",
            "properties":  map[interface {}]interface {}{
                "kty": map[interface {}]interface {}{
                    "description": "type",
                    "type":        "string",
                },
                "e": map[interface {}]interface {}{
                    "description": "key",
                    "type":        "string",
                },
                "use": map[interface {}]interface {}{
                    "description": "sign",
                    "type":        "string",
                },
                "alg": map[interface {}]interface {}{
                    "description": "algorithm",
                    "type":        "string",
                },
                false: map[interface {}]interface {}{
                    "description": "key",
                    "type":        "string",
                },
            },
        },
    },
}
```

But when I quote the n property like the yaml below

```yaml
definitions: 
  key:
    description: key content
    properties:
      kty:
        description: type
        type: string
      e:
        description: key
        type: string
      use:
        description: sign
        type: string
      alg:
        description: algorithm
        type: string
      'n':
        description: key
        type: string
```

I get: 

```go
map[interface {}]interface {}{
    "definitions": map[interface {}]interface {}{
        "key": map[interface {}]interface {}{
            "properties": map[interface {}]interface {}{
                "kty": map[interface {}]interface {}{
                    "description": "type",
                    "type":        "string",
                },
                "e": map[interface {}]interface {}{
                    "description": "key",
                    "type":        "string",
                },
                "use": map[interface {}]interface {}{
                    "description": "sign",
                    "type":        "string",
                },
                "alg": map[interface {}]interface {}{
                    "type":        "string",
                    "description": "algorithm",
                },
                "n": map[interface {}]interface {}{
                    "description": "key",
                    "type":        "string",
                },
            },
            "description": "key content",
        },
    },
}
```

**Comment by hallettj** [comment]
Interpreting the map keys `y` and `n` as boolean values is *not* intended behavior for YAML. This is a bug in the YAML parser that go-swagger uses.

For comparison, try using this online YAML parser with map keys like `n` and `y`: http://yaml-online-parser.appspot.com/

I also want to point out: I think it is misleading to include this bug in the list of "fixed bugs" in the release notes for go-swagger version 0.7.4. I came to look for a new release because I encountered the same issue. I saw that note saying that the issue was fixed. Then I was confused when my problem persisted even after updating.
